CMakeLists.txt

```diff
@@ -29,7 +29,11 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 include(GNUInstallDirs)
 
-add_subdirectory("power-grid-model")
+find_package(Eigen3 CONFIG REQUIRED)
+find_package(Boost REQUIRED)
+find_package(nlohmann_json CONFIG REQUIRED)
```

Comment:
not trying to revert this PR, but do you need to find these again? since the PGM core already hard-depends on this, you already have this from that location. Even more so, you're now actually overwriting the `Eigen3_FOUND`, `Boost_FOUND` and `nlohmann_json_FOUND` variables (and everything related). I think this is both redundant and actually more error-prone.